console/console-init/ui/src/modules/iot-device/components/MangeColumnModal/ManageColumnModal.tsx

```diff
@@ -0,0 +1,369 @@
+              />
+              <DataListItemCells
+                dataListCells={[
+                  <DataListCell
+                    id="device-list-column-management-status"
+                    key="device-list-column-management-status"
+                  >
+                    Status
+                  </DataListCell>
+                ]}
+              />
+            </DataListItemRow>
+          </DataListItem>
+          <DataListItem
+            id="device-list-col-mangmnt-last-updated"
+            aria-labelledby="device-list-column-management-last-updated"
+          >
+            <DataListItemRow>
+              <DataListCheck
+                id="device-list-col-mangmnt-last-updated-checkbox"
+                aria-labelledby="device-list-column-management-last-updated"
+                isChecked={isLastUpdatedChecked}
+                name="lastUpdated"
+                onChange={handleChange}
+              />
+              <DataListItemCells
+                dataListCells={[
+                  <DataListCell
+                    id="device-list-column-management-last-updated"
+                    key="device-list-column-management-last-updated"
+                  >
+                    Last updated
+                  </DataListCell>
+                ]}
+              />
+            </DataListItemRow>
+          </DataListItem>
+          <DataListItem
+            id="device-list-col-mangmnt-last-seen"
+            aria-labelledby="device-list-column-management-last-seen"
+          >
+            <DataListItemRow>
+              <DataListCheck
+                id="device-list-col-mangmnt-last-seen-checkbox"
+                aria-labelledby="device-list-column-management-last-seen"
+                isChecked={isLastSeenChecked}
+                name="lastSeen"
+                onChange={handleChange}
+              />
+              <DataListItemCells
+                dataListCells={[
+                  <DataListCell
+                    id="device-list-column-management-last-seen"
+                    key="device-list-column-management-last-seen"
+                  >
+                    Last seen
+                  </DataListCell>
+                ]}
+              />
+            </DataListItemRow>
+          </DataListItem>
+          <DataListItem
+            id="device-list-col-mangmnt-added-date"
+            aria-labelledby="device-list-column-management-added-date"
+          >
+            <DataListItemRow>
+              <DataListCheck
+                id="device-list-col-mangmnt-added-date-checkbox"
+                aria-labelledby="device-list-column-management-added-date"
+                isChecked={isAddedDateChecked}
+                name="addedDate"
+                onChange={handleChange}
+              />
+              <DataListItemCells
+                dataListCells={[
+                  <DataListCell
+                    id="device-list-column-management-added-date"
```

Comment:
Instead of repeating the same component structure, isolate the component in a method and pass 'type' 'date' etc as argument.

Reply:
> Instead of repeating the same component structure, isolate the component in a method and pass 'type' 'date' etc as argument.

yes, aggreed, It would better to create a function to render DataListItem if possible.